nym-vpn-x/src-tauri/src/commands/connection.rs

```diff
@@ -178,13 +178,14 @@ pub async fn disconnect(
     grpc: State<'_, Arc<GrpcClient>>,
 ) -> Result<ConnectionState, CmdError> {
     debug!("disconnect");
-    let mut app_state = state.lock().await;
+    let app_state = state.lock().await;
     if !matches!(app_state.state, ConnectionState::Connected) {
         return Err(CmdError::new(
             CmdErrorSource::CallerError,
             &format!("cannot disconnect from state {:?}", app_state.state),
         ));
     };
+    drop(app_state);
```

Comment:
I think having the scope track the ownership instead of explicit drop would be preferable

Reply:
Imho this is more than a style thing, resource ownership tracking scope is a relatively core Rust idiom with many advantages over explicit drop

Using scope it's more robust and less risk of mistakes, especially as the code evolves and with early returns

Reply:
Like you said it is your opinion, and I dont share it.
For example take a look at the official Mutex doc -> https://doc.rust-lang.org/std/sync/struct.Mutex.html

> To unlock a mutex guard sooner than the end of the enclosing scope, either create an inner scope or drop the guard manually.

```rust
// We drop the `data` explicitly because it's not necessary anymore and the
// thread still has work to do. This allows other threads to start working on
// the data immediately, without waiting for the rest of the unrelated work
// to be done here.
//
// It's even more important here than in the threads because we `.join` the
// threads after that. If we had not dropped the mutex guard, a thread could
// be waiting forever for it, causing a deadlock.
// As in the threads, a block could have been used instead of calling the
// `drop` function.
drop(data);
```

Where is it stated that one should "avoid" using drop because it's bad?  Or even is discouraged in favor to use inner scopes?

So when you said (tldr) "using `drop` is bad, it's not rust idiom", it's your own take.